File: src/go/dsa/myset.go
package dsa

// MySet is an interface which defines methods for sets.
type MySet[T comparable] interface {
	// Add inserts the given element to a set only if it was not already there, and
	// reports whether insertion was successful.
	Add(element T) bool

	// Clear empties a set of all elements.
	Clear()

	// Contains reports whether the given element is in a set.
	Contains(element T) bool

	// Equals reports whether obj is equal to a set.
	Equals(obj any) bool

	// IsEmpty reports whether a set has no elements.
	IsEmpty() bool

	// Remove attempts to delete the given element from a set, and reports whether
	// element was found.
	Remove(element T) bool

	// Size returns the number of elements in a set.
	Size() uint

	// ToArray returns an array containing all the elements in a set.
	ToArray() []T

	// ToString returns a string representation of a set, e.g.
	// "[element1, element2, element3, ..., elementN]".
	ToString() string
}

File: src/MySet.java
/**
 * Interface which defines methods for sets.
 *
 * @param <T> data type
 */
public interface MySet<T> {
    /**
     * Inserts an element into this set if it is not already there.
     *
     * @param element element to add
     * @return true if element was not present, false otherwise
     */
    public boolean add(T element);

    /**
     * Empties this set of all elements.
     */
    public void clear();

    /**
     * Checks if an element is in this set.
     *
     * @param element element to check for
     * @return true if found, false otherwise
     */
    public boolean contains(Object element);

    /**
     * Compares an object with this set for equality.
     *
     * @param object object to compare to this set
     * @return true if object and this set are equal
     */
    @Override
    public boolean equals(Object object);

    /**
     * Checks if this set has no elements.
     *
     * @return true if this set is empty, false otherwise
     */
    public boolean isEmpty();

    /**
     * Removes an element from this set.
     *
     * @param element element to remove
     * @return true if element found, false otherwise
     */
    public boolean remove(Object element);

    /**
     * Returns the number of elements in this set.
     *
     * @return size of set
     */
    public int size();

    /**
     * Returns an array containing all the elements in this set.
     *
     * @return array of set elements
     */
    public Object[] toArray();

    /**
     * Returns a string representation of this set, e.g. "[element1, element2, element3, ..., elementN]".
     *
     * @return string form of this set
     */
    @Override
    public String toString();
}
